src/parser.cpp

```diff
@@ -1,18 +1,19 @@
+#include "ada/parser.h"
+
+#include <limits>
+
 #include "ada.h"
-#include "ada/common_defs.h"
 #include "ada/character_sets-inl.h"
+#include "ada/common_defs.h"
+#include "ada/log.h"
 #include "ada/unicode.h"
 #include "ada/url-inl.h"
-#include "ada/log.h"
-#include "ada/parser.h"
-
-#include <limits>
 
 namespace ada::parser {
 
-template <class result_type>
-result_type parse_url(std::string_view user_input,
-                      const result_type* base_url) {
+template <class result_type, bool store_values>
+result_type parse_url_impl(std::string_view user_input,
```

Comment:
Can we keep the function name same? This is a breaking change.

Reply:
I actually kept the old function `parse_url`, as Daniel suggested. 
![image](https://github.com/ada-url/ada/assets/40635471/b53e1821-fb27-4750-b34a-785acf6622e6)

Now the old `parse_url` calls the new one `parse_url_impl`, with the `store_values` as `true`.